torch/_decomp/__init__.py

```diff
@@ -250,13 +252,170 @@ def remove_decompositions(
 #
 # list was copied from torch/_inductor/decomposition.py
 # excluding decompositions that results in prim ops
 # Resulting opset of decomposition is core aten ops
 def core_aten_decompositions() -> Dict[torch._ops.OperatorBase, Callable]:
+    decomp_table = _core_aten_decompositions_post_autograd()
+
+    # If it is fbcode change, we return the old decomposition list
+    from torch._inductor import config
+
+    if config.is_fbcode():
+        return decomp_table
+
+    # We are deleting custom decomp in core_aten_decomp
+    # for CIA ops but it should be fine technically
+    # because this table is only meant to be used in export context
+    # in which we really carefully control the decomp behaviour
+    # In any case, C++ decomps should be preferred
+    cia_ops_that_should_be_removed = [
+        aten.all.dimname,
+        aten.index_add.dimname,
+        aten.index_copy.dimname,
+        aten.index_fill.Dimname_Scalar,
+        aten.index_fill.Dimname_Tensor,
+        aten.norm.names_ScalarOpt_dim_dtype,
+        aten.norm.names_ScalarOpt_dim,
+        aten.silu_backward.default,
+        aten.std.default,
+        aten.std.dim,
+        aten.std.names_dim,
+        aten.std.correction_names,
+        aten.std_mean.default,
+        aten.std_mean.dim,
+        aten.std_mean.names_dim,
+        aten.std_mean.correction_names,
+        aten.upsample_bilinear2d.vec,
+        aten.upsample_trilinear3d.vec,
+    ]
+
+    for k in list(decomp_table.keys()):
+        if k in cia_ops_that_should_be_removed:
+            del decomp_table[k]
+
+    for op in _collect_all_valid_cia_ops():
+        # [NOTE] Seperating out func.decompose
+        # Ideally we should be able to just register func.decompose but
+        # we can't as this decomp is gonna be registered to the py_impl.
+        # As a result it will infinitely recurse. So we first check if the op
+        # has py_impl entry for CIA and if it is we use that first. If not,
+        # we register C++ query to py_impl.
+        if (
+            torch._C.DispatchKey.CompositeImplicitAutograd in op.py_kernels
+        ) and not isinstance(
+            op.py_kernels[torch._C.DispatchKey.CompositeImplicitAutograd],
+            torch._C.DispatchKey,
+        ):
+            decomp_table[op] = op.py_kernels[
+                torch._C.DispatchKey.CompositeImplicitAutograd
+            ]
+        else:
+            decomp_table[op] = partial(_is_special_op_to_decompose, kernel=op)
+    return decomp_table
+
+
+def decomp_table_to_post_autograd_aten():
+    decomp_table = {}
+    for k in _collect_all_valid_cia_ops():
+        if torch._C.DispatchKey.CompositeImplicitAutograd in k.py_kernels:
+            decomp_table[k] = k.py_kernels[
+                torch._C.DispatchKey.CompositeImplicitAutograd
+            ]
+        else:
+            decomp_table[k] = partial(_is_special_op_to_decompose, kernel=k)
+    return decomp_table
+
+
+def _core_aten_decompositions_post_autograd() -> (
```

Comment:
This basically will return the old core_aten_decomp table.

Reply:
is the idea that this is just temporary and we want to kill it later? (and rationalize the cases where people actually rely on this specific decomp list)

Reply:
Yep temporary.